esque/controller/topic_controller.py

```diff
@@ -119,23 +130,34 @@ def get_offsets_closest_to_timestamp(
             topic_partition.partition: topic_partition.offset for topic_partition in topic_partitions_with_new_offsets
         }
 
-    def update_from_cluster(self, topic: Topic, *, retrieve_last_timestamp: bool = False) -> Topic:
+    def update_from_cluster(
+        self, topic: Topic, *, retrieve_last_timestamp: bool = False, retrieve_partition_data: bool = True
+    ) -> Topic:
         """Takes a topic and, based on its name, updates all attributes from the cluster"""
 
-        topic.partition_data = self._get_partition_data(topic, retrieve_last_timestamp)
+        if retrieve_partition_data:
+            topic.partition_data = self._get_partitions(topic, retrieve_last_timestamp)
+        else:
+            topic.partition_data = self._get_partitions(topic, retrieve_last_timestamp, get_partition_data=False)
         topic.config = self.cluster.retrieve_config(ConfigResource.Type.TOPIC, topic.name)
 
         topic.is_only_local = False
 
         return topic
 
-    def _get_partition_data(self, topic: Topic, retrieve_last_timestamp: bool) -> List[Partition]:
-
+    def _get_partitions(
+        self, topic: Topic, retrieve_last_timestamp: bool, get_partition_data: bool = True
```

Comment:
Should there be an assertion to make sure this isn't called with `retrieve_last_timestamp=True` and `get_partition_data=False`?

Reply:
sounds like a good idea. will add that.

Reply:
Done.